src/app/api/Requirements.ts

```diff
@@ -0,0 +1,29 @@
+export class Tag {
+    type: string
+    value: string
+
+    constructor(value: string) {
+        this.type = "tag"
+        this.value = value
+    }
+}
+
+export class Branch {
+    type: string
+    value: string
+
+    constructor(value: string) {
+        this.type = "branch"
+        this.value = value
+    }
+}
+
+export class Source {
+    tags: Tag[]
+    branches: Branch[]
+
+    constructor(tags: Tag[], branches: Branch[]) {
+        this.tags = tags
+        this.branches = branches
+    }
+}
```

Comment:
All of these seem to be intending to reflect shape more than logic, so IMO, instead of classes you should use `interface`s or `type` aliases. TypeScript type inference is good enough so when you type an object literal, it will know what type it is.

I'd also moved these under a common source, i.e. the `models` directory. There, you even have proper `interface`s to declare types in the `repository.ts` file.

Reply:
Yep, I thought about that, but I changed coming from a Toma's comment. The point is, if I declare it as `type` I only can create object with these requirements using the `{key: value, key:value}` form?

Reply:
And yes, all of them are `models` no dependencies, no logic. Just plain objects.

Reply:
Example:

```
export interface Tag {
  type?: "tag";
  value: string
}
```

Usage:

```
    public async tags(owner: string, repo: string): Promise<Tag[]> {
        const values = await this.focusName(`/repos/${owner}/${repo}/tags`)
        return values.map(it => ({ value: it }))
    }
```

Returned values are of type `Tag` as the function is type to return an array of that type, and you are properly setting an object of that shape.

Reply:
In general, classes in JS/TS are a bit misleading, so I'd advise to use them sparingly.

Reply:
Sure! I mean to do not less the context of the type, but yep, I will update it. Thanks.

Reply:
Sorry, I answered you with the examples without reading your response as GitHub didn't update them, weird.

Anyway, just my two cents on that, but it would be ok if you let it as it was :+1:

I can be more grumpy :sweat_smile:, but I guess that for the scope of this project there's no need to go further, but let me know otherwise. Project is a cool idea by the way! :raised_hands:

Reply:
I really appreciate your comments, and to be honest, it is my first time using typescript, so no idea what is better hehe! I have just updated, sorry for the delay.

Reply:
Feel free to add any extra comment!